test/helpers/helpers.go

```diff
@@ -220,6 +225,27 @@ func UpdateRuntimeObject(client test.FrameworkClient, obj runtime.Object) error
 	)
 }
 
+func UpdateCatalogSourceWithRetries(client test.FrameworkClient, obj *olmv1alpha1.CatalogSource) error {
+	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		err := client.Update(context.TODO(), obj)
+		if err != nil && apierrors.IsConflict(err) {
+			logrus.Infof("Ran into a conflict while updating the %s CatalogSource - retrying...", obj.Name)
+			clone := obj.DeepCopyObject().(*olmv1alpha1.CatalogSource)
+			key := types.NamespacedName{
+				Name:      obj.Name,
+				Namespace: obj.Namespace,
+			}
+			err = client.Get(context.TODO(), key, clone)
+			if err != nil {
+				return err
+			}
+
+			obj.SetResourceVersion(clone.GetResourceVersion())
+		}
+		return err
+	})
+}
```

Comment:
Open to suggestions on this implementation and whether there's something easier to implement without mixing client-go and controller-runtime mechanisms. This was mainly needed to cut down on flakes as we don't have any retry on conflict mechanism when we're attempting to update any of the APIs or CatalogSource resources that we manage.